coediting/hooks/index.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * External dependency
+ */
+import classnames from 'classnames';
+import { connect } from 'react-redux';
+
+/**
+ * WordPress dependency
+ */
+import { BlockEdit, getBlockDefaultClassname, getBlockType, hasBlockSupport } from '@wordpress/blocks';
+import { addFilter } from '@wordpress/hooks';
+import { getWrapperDisplayName } from '@wordpress/element';
+
+/**
+ * Internal dependency
+ */
+import './style.scss';
+// TODO: Move selectors to the local folder
+import {
+	getBlock,
+	getFrozenBlockCollaboratorColor,
+	getFrozenBlockCollaboratorName,
+	isBlockFrozenByCollaborator,
+} from '../../editor/selectors';
+
+const withFrozenMode = ( OriginalComponent, originalProps ) => {
+	const Component = ( { isFrozenByCollaborator, ...props } ) => {
+		if ( isFrozenByCollaborator ) {
+			const { block, collaboratorColor, collaboratorName } = props;
+			const { attributes, name, isValid, uid } = block;
+			const blockType = getBlockType( name );
+
+			// Determine whether the block has props to apply to the wrapper.
+			let wrapperProps;
+			if ( blockType.getEditWrapperProps ) {
+				wrapperProps = blockType.getEditWrapperProps( attributes );
+			}
+
+			// Generate a class name for the block's editable form
+			const generatedClassName = hasBlockSupport( blockType, 'className', true ) ?
```

Comment:
@aduth @youknowriad can we move it to `EditBlock`? I think it's internal thing which shouldn't be part of this and the original component.

Reply:
Moved in https://github.com/WordPress/gutenberg/pull/4009.